### DISCUSSION TITLE
Can not commit through command line in Mac

### DISCUSSION THREAD

**Discussion by sleepy-analyst** [community/community#133925] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi everyone, 

Whenever I try to - 

git push -u origin main
asks for user id - entered OK
asks for Password - entered name of Create Personal Access Token

remote: Support for password authentication was removed on August 13, 2021. remote: 
Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication. fatal: Authentication failed for 'https://github.com/Wilko500/PrimroseRepository.git/' 

I keep getting error even if I type in the correct password.

**Comment by subhanjandas** [comment, marked as answer]
It sounds like you're facing an issue with authenticating to GitHub from your Mac using Git. This is a common issue, especially since GitHub removed support for password authentication for Git operations in August 2021. Now, you need to use a Personal Access Token (PAT) instead of your GitHub password.

### Here's the Steps to Push Your Project to GitHub Using a Personal Access Token

#### 1. Create a Personal Access Token (PAT)
1. Go to [GitHub](https://github.com).
2. Click on your profile picture in the top right, then click on **Settings**.
3. In the left sidebar, click **Developer settings**.
4. Click **Personal access tokens** and then **Tokens (classic)**.
5. Click **Generate new token**.
6. Give your token a descriptive name, select the scopes you need (at least `repo` to push code), and generate the token.
7. **Copy the token** and save it in a secure place, as you won't be able to see it again.

#### 2. Configure Git to Use the Personal Access Token

Since you're using macOS, you can use the Git credential helper to store your token securely.

Open your terminal and run:
```sh
git config --global credential.helper osxkeychain
```

#### 3. Clone Your Repository

If you haven't already cloned your repository, you should do so. It's important to use the HTTPS URL.

```sh
git clone https://github.com/UserId/projectID.git
```

#### 4. Navigate to Your Project Directory

```sh
cd projectID
```

#### 5. Initialize Git in Your Project Directory (if not already done)
If you haven't initialized your Git repository, do it now:

```sh
git init
git branch -m main
git add .
git commit -m "Initial commit"
```

#### 6. Add the Remote Repository
Ensure you add the correct remote URL:

```sh
git remote add origin https://github.com/UserId/projectID.git
```

#### 7. Push Your Changes Using PAT
When you push your changes, you'll be prompted for your username and password. Enter your GitHub username and use the Personal Access Token as your password.

```sh
git push -u origin main
```
When prompted for a password, paste your Personal Access Token.

Thanks :)

**Reply by sleepy-analyst** [reply]
Thanks, typing in the PAT instead of password worked!
